### PR TITLE
Refs [TASK] [37455] Tạo giao diện màn hình danh sách hình ảnh

### PR DESCRIPTION
**Pull request by luongth-2520**
## Related Tickets
- [#Task 37455: Tạo giao diện màn hình danh sách hình ảnh](https://edu-redmine.sun-asterisk.vn/issues/37455)
## WHAT
- Tạo giao diện màn hình danh sách hình ảnh
## Evidence (Screenshot or Video)
Screenshot from 2021-06-22 14-27-59

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/unsplash03/screen/home/HomeFragment.kt [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            }
+            else -> Unit
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+    companion object {
+        private const val HOME = 0
+        private const val COLLECTION = 1
+        private const val FAVORITE = 2
```
Comment (khoaqh-2470) [inline app/src/main/java/com/sun/unsplash03/screen/home/HomeFragment.kt]: suggest enum class

File: app/src/main/java/com/sun/unsplash03/screen/main/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+class MainActivity : BaseActivity<ActivityMainBinding, MainViewModel>() {
+
+    override val viewModel: MainViewModel by viewModel()
+
+    override fun inflateViewBinding(inflater: LayoutInflater) = ActivityMainBinding.inflate(inflater)
+
+    override fun initView() {
+        viewBinding.lifecycleOwner = this
+        viewBinding.viewModel = viewModel
```
Comment (khoaqh-2470) [inline app/src/main/java/com/sun/unsplash03/screen/main/MainActivity.kt]: scope function

File: app/src/main/java/com/sun/unsplash03/screen/photo/PhotoFragment.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+class PhotoFragment : BaseFragment<FragmentPhotoBinding, PhotoViewModel>() {
+
+    override val viewModel: PhotoViewModel by viewModel()
+
+    override fun inflateViewBinding(inflater: LayoutInflater)= FragmentPhotoBinding.inflate(inflater)
+
+    override fun setUpView() {
+        viewBinding.lifecycleOwner = this.viewLifecycleOwner
+        viewBinding.viewModel = viewModel
```
Comment (khoaqh-2470) [inline app/src/main/java/com/sun/unsplash03/screen/photo/PhotoFragment.kt]: above

File: app/src/main/java/com/sun/unsplash03/screen/home/HomeFragment.kt [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+class HomeFragment : BaseFragment<FragmentHomeBinding, HomeViewModel>() {
+
+    override val viewModel: HomeViewModel by viewModel()
+
+    override fun inflateViewBinding(inflater: LayoutInflater) =
+        FragmentHomeBinding.inflate(inflater)
+
+    override fun setUpView() {
+        viewBinding.lifecycleOwner = this.viewLifecycleOwner
+        viewBinding.viewModel = viewModel
```
Comment (khoaqh-2470) [inline app/src/main/java/com/sun/unsplash03/screen/home/HomeFragment.kt]: scope function

File: app/src/main/java/com/sun/unsplash03/screen/collection/CollectionFragment.kt [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+class CollectionFragment : BaseFragment<FragmentCollectionBinding, CollectionViewModel>() {
+
+    override val viewModel: CollectionViewModel by viewModel()
+
+    override fun inflateViewBinding(inflater: LayoutInflater) = FragmentCollectionBinding.inflate(inflater)
+
+    override fun setUpView() {
+        viewBinding.lifecycleOwner = this.viewLifecycleOwner
+        viewBinding.viewModel = viewModel
```
Comment (khoaqh-2470) [inline app/src/main/java/com/sun/unsplash03/screen/collection/CollectionFragment.kt]: scope function

**Review by trinq-2471 — APPROVED** [review]

File: app/src/main/res/menu/menu_search.xml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
+    <item
+        android:id="@+id/menuSearch"
```
Comment (tuanlvt-1117) [inline app/src/main/res/menu/menu_search.xml]:
```suggestion
        android:id="@+id/searchItem"
```

File: app/src/main/res/menu/menu_bottom_navigation.xml [hunk 7]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android">
+    <item
+        android:id="@+id/menuHome"
```
Comment (tuanlvt-1117) [inline app/src/main/res/menu/menu_bottom_navigation.xml]: tương tự
```suggestion
        android:id="@+id/homeItem"
```

File: app/src/main/res/layout/fragment_photo.xml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    </data>
+
+    <androidx.swiperefreshlayout.widget.SwipeRefreshLayout
+        android:id="@+id/swipeRefreshMain"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content">
+
+        <androidx.constraintlayout.widget.ConstraintLayout
+            android:layout_width="match_parent"
+            android:layout_height="match_parent">
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/fragment_photo.xml]: layout dư

File: app/src/main/res/layout/fragment_favorite.xml [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+<layout>
+
+    <data>
+
+        <variable
+            name="viewModel"
+            type="com.sun.unsplash03.screen.favorite.FavoriteViewModel" />
+    </data>
+
+    <androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/fragment_favorite.xml]:
```suggestion
<?xml version="1.0" encoding="utf-8"?>
<layout xmlns:android="http://schemas.android.com/apk/res/android">

    <data>

        <variable
            name="viewModel"
            type="com.sun.unsplash03.screen.favorite.FavoriteViewModel" />
    </data>

    <androidx.constraintlayout.widget.ConstraintLayout
```

File: app/src/main/res/layout/fragment_collection.xml [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+
+    <data>
+
+        <variable
+            name="viewModel"
+            type="com.sun.unsplash03.screen.collection.CollectionViewModel" />
+    </data>
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/fragment_collection.xml]:
```suggestion
<?xml version="1.0" encoding="utf-8"?>
<layout xmlns:android="http://schemas.android.com/apk/res/android">

    <data>

        <variable
            name="viewModel"
            type="com.sun.unsplash03.screen.collection.CollectionViewModel" />
    </data>

    <androidx.constraintlayout.widget.ConstraintLayout
```

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**